apps/prairielearn/src/pages/workspaceLogs/workspaceLogs.ts

```diff
@@ -116,7 +115,7 @@ async function loadLogsForWorkspaceVersion(
 // if the instructor has access to the workspace (i.e., course instance student
 // data view permission, or access to a workspace owned by the user); that's
 // already been checked by the workspace authorization middleware.
-router.use(authzHasCoursePreviewOrInstanceView);
+router.use((await import('../../middlewares/authzHasCoursePreviewOrInstanceView.js')).default);
```

Comment:
Not that it really matters, but I prefer non-dynamic imports whenever it makes sense. The only place I think we need dynamic imports is in `server.ts` (where we don't want to have to name 100s of imports) and when they need to be conditional, e.g. with a `isEnterprise()` check.

In other words, just change the import to this:

```ts
import authzHasCoursePreviewOrInstanceView from '../../middlewares/authzHasCoursePreviewOrInstanceView.js';
```

It's horrible that the types between the middleware and function versions are both compatible with `router.use`! We could resolve that in this case by dropping `req` from the signature of the non-middleware function, as it's unused.

Reply:
> In other words, just change the import to this:
> 
> ```ts
> import authzHasCoursePreviewOrInstanceView from '../../middlewares/authzHasCoursePreviewOrInstanceView.js';
> ```

Given that there is a named import with that name, wouldn't this be confusing? Someone reading the code might mistake the default export for the named export.

Reply:
I buy that. I'll land this as-is 👍